packages/design-system/src/components/TextField/Mask.test.tsx

```diff
@@ -1,153 +1,131 @@
-    wrapper
-      .find('input')
-      .props()
-      .onBlur({ target: { value: '1234' }, persist: jest.fn() });
-
-    expect(onBlur.mock.calls.length).toBe(1);
+    renderMask({ mask: 'currency' }, { value: '123', onBlur });
+    userEvent.click(getInput());
+    userEvent.tab();
+    expect(onBlur).toHaveBeenCalledTimes(1);
   });
 
   it('calls onChange', () => {
     const onChange = jest.fn();
-    const wrapper = render({ mask: 'currency' }, { value: '123', onChange: onChange }).wrapper;
-    const input = wrapper.find('input');
-
-    input.simulate('change', { target: { value: '123' } });
-
-    expect(onChange.mock.calls.length).toBe(1);
+    renderMask({ mask: 'currency' }, { value: '123', onChange });
+    userEvent.type(getInput(), 'hello');
+    expect(onChange).toHaveBeenCalledTimes('hello'.length);
   });
 
   it('changes to a controlled field using defaultValue', () => {
-    const data = render({ mask: 'currency' }, { defaultValue: '1234' });
-    const input = data.wrapper.find('input');
-
-    expect(input.prop('value')).toBe('1,234');
+    renderMask({ mask: 'currency' }, { defaultValue: '1234' });
+    expect(getInput().value).toBe('1,234');
   });
 
   describe('Controlled component behavior', () => {
     it('will not cause masking until blur when value prop still matches unmasked input', () => {
-      const onChange = (event) => {
-        // Simulate the change bubbling up to the controlling component and the
-        // controlling component then updating the value prop.
-        this.setProps({
-          children: (
-            <input name="foo" type="text" value={unmaskValue(event.target.value, 'currency')} />
-          ),
-        });
-      };
-      const { wrapper } = render({ mask: 'currency', onChange }, { value: '1000' });
-      const input = () => wrapper.find('input');
-
-      expect(input().prop('value')).toBe('1,000');
-
-      input()
-        .props()
-        .onChange({ target: { value: '1,0000' } });
-      expect(input().prop('value')).toBe('1,0000');
-
-      input()
-        .props()
-        .onBlur({ target: { value: '1,0000' }, persist: jest.fn() });
-      expect(input().prop('value')).toBe('10,000');
+      // Simulate the change bubbling up to the controlling component and the
+      // controlling component then updating the value prop.
+      function ControlledComponentParent() {
+        const [text, setText] = useState('1000');
+        return (
+          <Mask mask="currency">
+            <input
+              name="foo"
+              type="text"
+              value={text}
+              onChange={(event) => setText(unmaskValue(event.target.value, 'currency'))}
+            />
+          </Mask>
+        );
+      }
+
+      render(<ControlledComponentParent />);
+      expect(getInput().value).toBe('1,000');
+
+      // Remove the comma
+      userEvent.type(getInput(), '{arrowleft}{arrowleft}{arrowleft}{backspace}');
```

Comment:
Oh wow didn't know about this either.

Reply:
It feels like a cheat code 😆 `^^vv<><>...`